src/characters.rs

```diff
@@ -60,3 +61,41 @@ pub async fn get_characters(
         HttpResponse::BadRequest().json("no")
     }
 }
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NewAssignment {
+    actor_id: Option<i32>,
+    character_id: i32,
+}
+
+#[post("/characters/{script_id}/assignCharacter")]
+pub async fn assign_character(
+    data: web::Data<AppState>,
+    script_id: web::Path<i32>,
+    new_assignment: web::Json<NewAssignment>,
+) -> impl Responder {
+    // actor_id is an Option because null is passed when a character is being unassigned
+    let NewAssignment {
+        actor_id,
+        character_id,
+    } = new_assignment.into_inner();
+    let script_id = script_id.into_inner();
+    let result = sqlx::query!(
+        "UPDATE characters
+    SET actor_id = $1
+    WHERE script_id = $2 AND id = $3
+    RETURNING *",
+        actor_id,
+        script_id,
+        character_id
+    )
+    .fetch_one(&data.db)
+    .await;
+
+    if result.is_ok() {
+        HttpResponse::Ok().json(true)
+    } else {
+        HttpResponse::BadRequest().json("no")
```

Comment:
The json("no") was just some nonsense I was using for testing. That shouldn't be copied as a good practice to other files. We should probably return just a 500 on failure.
```rust
HttpResponse::BadRequest().finish()
```

Reply:
Ah, gotcha. Makes sense. I went with InternalServerError since that is the 500 and is consistent with what we've been using historically.